FreeRTOS/FreeRTOS-Plus-TCP
# Run uncrustify before applying complexity

The GNU application "complexity" analyses source files and gives a complexity score to each of the procedures / functions found. A score of 9 or lower is considered good.

But unfortunately, "complexity" stops parsing a source file as soon as it sees a function that is indented either with spaces or with tabs:
~~~c
    int test_2( int value )
    {
        if( value > 0 )
        {
            printf( "test_2: value = %d\n", value );
        }
    }
~~~

It seems to help if the source code is reformatted to remove indentation. I added a new config file for uncrustify called tools/uncrustify.complexity.cfg, which is used by a script called ".github\workflows\ci.yml".

Test Steps
-----------
Add the function here above to a source file and check it with complexity.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
